### ISSUE TITLE
non ascii chars in field values causing csvcut to create new fields

### ISSUE THREAD

**Issue by bbobbadi** [wireservice/csvkit#631]
[testNonASCII.zip](https://github.com/wireservice/csvkit/files/339891/testNonASCII.zip)

Hi There, 
I have been using csvcut tool. 
My data is in csv with combination of ascii and non ascii characters. The issue I am facing is when there are some non ascii chars in the filed value it is causing new fields and new lines. Could you please help me to handle non ascii chars. 
I have attached the input and output files. The command I have used is " csvcut -c Col4,preview,Col2,searchable_body,Col3 testNonASCII > result.csv"
I need to carry these non-ascii chars to the output file with out changes. 

Thanks in advance for the help.

**Comment by jpmckinney** [comment]
Have you tried with the latest version from GitHub?

    pip install -e git+git://github.com/onyxfish/csvkit.git@master#egg=csvkit

**Comment by bbobbadi** [comment]
I got the error "Command python setup.py egg_info failed with error code 1 in /tmp/pip-build-adaptive/dbfread" when I tried "pip install -e git+git://github.com/onyxfish/csvkit.git@master#egg=csvkit".

Attached full log. 

[pip.txt](https://github.com/wireservice/csvkit/files/340210/pip.txt)

**Comment by jpmckinney** [comment]
What version of Python are you running?

    python --version

**Comment by bbobbadi** [comment]
2.6.6

**Comment by jpmckinney** [comment]
The latest version of csvkit only supports 2.7 and higher.

**Comment by bbobbadi** [comment]
Is there a way I can have python 2.7  and use it for csvkit on same machine where python 2.6.6 is default?

**Comment by jpmckinney** [comment]
If you install Python 3, it should not clash with Python 2. Then, you can use [virtualenv](https://virtualenv.pypa.io/en/stable/) to use csvkit in Python 3.

**Comment by bbobbadi** [comment]
I have tried csvkit latest with Python 2.7.12 on virtualenv, but the result is same as before.

**Comment by jpmckinney** [comment]
I can't reproduce. Try:

    pip install --upgrade -e git+git://github.com/onyxfish/csvkit.git@master#egg=csvkit

Also, I get different output when using the newest version of csvkit, so I think the issue is fixed.

**Comment by bbobbadi** [comment]
(myPython2.7)[me@mysys ~]$ python -V
Python 2.7.12
(myPython2.7)[me@mysys ~]$ rm result.csv
(myPython2.7)[me@mysys ~]$ csvcut -c Col4,preview,Col2,searchable_body,Col3 testNonASCII.csv > result.csv
Output:
(myPython2.7)[me@mysys ~]$ cat result.csv
Col4,preview,Col2,searchable_body,Col3
cool4,,,3 accounting at everyone.�,
�,,,,
 hi in info mbi... mbi.instructure.com more of the week ,0,cool3,ALLOWED,

Input:
(myPython2.7)[me@mysys ~]$ cat testNonASCII.csv
Col4,searchable_body,result,sms_message_type,preview,subscriber_reputation,Col1,Col2,Col3
cool4,3 accounting at everyone.�� hi in info mbi... mbi.instructure.com more of the week ,ALLOWED,none,"Hi everyone.��CR_2x1$1x4_CR
Week 3 of accounting in the MBI...CR_2x1$1x4_CR
CR_2x1$1x4_CR
More info at mbi.instructure.comCR_2x1$1x4_CR
 ",0,cool1,cool2,cool3

Could you please point me where I am doing wrong here. I have ran "pip install --upgrade -e git+git://github.com/onyxfish/csvkit.git@master#egg=csvkit" before this test. The output is creating 3 rows for the input of 1 record.

**Comment by jpmckinney** [comment]
I get one row:

```
Col4,preview,Col2,searchable_body,Col3
cool4,"Hi everyone.  CR_2x1$1x4_CR
Week 3 of accounting in the MBI...CR_2x1$1x4_CR
CR_2x1$1x4_CR
More info at mbi.instructure.comCR_2x1$1x4_CR
 ",cool2,3 accounting at everyone.   hi in info mbi... mbi.instructure.com more of the week ,cool3
```

**Comment by jpmckinney** [comment]
Seems to be an issue in Python 2, but not Python 3.

**Comment by jpmckinney** [comment]
@onyxfish @nbedi Any idea of UTF-8 issues in Python 2?

**Comment by nbedi** [comment]
Possibly related to: https://github.com/wireservice/csvkit/issues/596 and https://github.com/wireservice/agate/issues/624 but it's a different kind of error.

I'll try to look into these this week.

**Comment by bbobbadi** [comment]
Thank you very much for the support. 
Just wanted to update you with my findings over the issue (thought of useful to fix the issue).

Found that "e280 a8e2 80a8" hex string  causing new record when it goes through csvcut.
I need to retain the special chars as it is in the input inside the field after csvcut output.

Any idea of such UTF-8 char-set may create new lines in csvcut? so that I can replace with constant before csvcut and replace back the constants with original char-set.

**Comment by jpmckinney** [comment]
The issue is with `\u2028` (LINE SEPARATOR) in Python 2, which now causes:

    Your file is not "utf-8" encoded. Please specify the correct encoding with the -e flag. Use the -v flag to see the complete error.

**Comment by onyxfish** [comment]
Fixed in csvkit `1.0.0`
